Dropdown Menu for Language

Yashs911 opened the pull request:
<!-- What issue does this PR close? -->
Added a dropdown menu for the Language feature.
### Screenshot
<!-- If this PR touches UI, please post evidence (screenshots) of it behaving correctly. -->
Desktop:
![image](https://user-images.githubusercontent.com/64412143/115721766-f922d500-a39b-11eb-98fe-41f3b6d34b3e.png)
Mobile:
![image](https://user-images.githubusercontent.com/64412143/115721942-21123880-a39c-11eb-9456-c3389cee6ee8.png)

### Stakeholders
<!-- @ tag stakeholders of this bug -->
@mekarpeles

Yashs911 commented:
Failing the following tests:
```
FAIL  static/build/page-book.css: 9.77KB > maxSize 9.6KB (gzip) 

 FAIL  static/build/page-edit.css: 20.16KB > maxSize 20KB (gzip) 

 FAIL  static/build/page-form.css: 20.18KB > maxSize 20KB (gzip)
```

jdlrobson commented:
While I haven't pulled down the code and looked at this, I really like the direction here. As we modernize the UI, I am trying to keep the client as light as possible. My intention is that we will lean on JavaScript only for workflows such as the lists feature and editor. These will be completely Vue based.

Navigation is something that should work without JavaScript, for a variety of reasons, but mainly because HTML is more future proof because of its backwards compatible nature. Right now, we're currently grappling with upgrading jQuery UI experiences that if taken away would break key components of the site, and I'm keen to avoid that in essential parts of the user interface like this one. I'd also like us to move away from jQuery and show/hide patterns so we don't get stuck into jQuery in the same way.

I am in the process of putting together a document explaining a roadmap for the frontend and a guiding philosophy but the approach outlined here definitely sits in well with that! Thanks for trying out different ideas. This one seems to fit in nicely! I urge you to continue it. Experimenting with language and then applying it to all over menus once proved seems like a great approach here.

jdlrobson commented on openlibrary/templates/lib/nav_foot.html:
```diff
@@ -48,17 +48,23 @@ <h2>$:_('Help')</h2>
+          </summary> 
+          <div class="language-dropdown-component">
+            <ul class="dropdown-menu" id="locale-options">
+              <li><a href="#" lang="cs" data-lang-id="cs" title="$_('Czech')">Čeština (cs)</a> </li>
```
INstead of # as the href here, could we use something that toggles the language? e.g. ?lang=cs

We can disable the link behaviour with JavaScript using a click handler with ev.preventDefault

Yashs911 replied:
@jdlrobson It's functional but you will not notice it on localhost. `HTTP_LANG` Cookie will be added and its value will change accordingly.

Yashs911 replied:
We can also go ahead with your suggestion.

Yashs911 replied:
The feature uses the following js to set the cookie.
https://github.com/internetarchive/openlibrary/blob/133ee89ae2bbb290d4a805b2be4763906aa06765/openlibrary/plugins/openlibrary/js/ol.js#L159

> I also think it /should/ do something locally to support our testing and development, even if it's not what we do in production. Otherwise the feature will look broken and be confusing. A query string parameter seems like a good fallback regardless.

I completely agree with you but even localhost:8080?lang=fr doesn't show any visual changes.

Yashs911 replied:
https://user-images.githubusercontent.com/64412143/117581175-2ea81c00-b119-11eb-8373-ce6ed287b5c0.mp4

jdlrobson reviewed (changes requested):
Looks great but doesn't appear to be functional yet.

jdlrobson commented:
> I completely agree with you but even localhost:8080?lang=fr doesn't show any visual changes.

I know, we'd need a backend change to support that, but I see that as a necessary part of this solution.

Yashs911 commented:
@jdlrobson What should be my next step? I will add the query string parameter. Anything else that I need to do?
Apart from that, if the dropdown is fine, I will start working on the header dropdown PR.

jdlrobson commented:
@mekarpeles can you advise how i18n should work without javascript, cookies disabled or for local development?

Yashs911 commented:
If we merge #5367 then we can remove all the changes related to the mobile view and remove most of the duplicate code.

mekarpeles commented:
@Yashs911 I think we may be able to move forward with this one since https://github.com/internetarchive/openlibrary/pull/5367 is merged

Yashs911 commented:
> @Yashs911 I think we may be able to move forward with this one since #5367 is merged
> 
> It looks like some conflicts need to be resolved

@mekarpeles I have fixed the merge conflict and made the necessary fixes.

I have also removed all the changes from the footer section for this PR as now we have IA Bar on Mobile, So do let me know how we want to proceed with the Footer section language list (completely removing it or merging PR 5367) because as you mentioned #5367 is **not merged** yet

Yashs911 commented:
Jest: "global" coverage threshold for statements (19%) not met: 18.81%
Jest: "global" coverage threshold for lines (19%) not met: 18.[86](https://github.com/internetarchive/openlibrary/runs/5055244081?check_suite_focus=true#step:11:86)%
Jest: "global" coverage threshold for functions (15%) not met: 14.81%

If someone can guide me on how to fix the above error and pre-commit error

mekarpeles commented on openlibrary/templates/site/alert.html:
```diff
@@ -4,8 +4,26 @@
+              <li><a href="#" lang="en" data-lang-id="en" title="$_('English')">English (en)</a></li>
+              <li><a href="#" lang="es" data-lang-id="es" title="$_('Spanish')">Español (es)</a></li>
+              <li><a href="#" lang="fr" data-lang-id="fr" title="$_('French')">Français (fr)</a></li>
+              <li><a href="#" lang="hr" data-lang-id="hr" title="$_('Croatian')">Hrvatski (hr)</a></li>
```
```suggestion
              <li><a href="#" lang="hr" data-lang-id="hr" title="$_('Croatian')">Hrvatski (hr)</a></li>
              <li><a href="#" lang="zh" data-lang-id="zh" title="$_('Chinese')">中文 (zh)</a></li>
```
